### PR TITLE
developer-guide: update guest kernel image section

### PR DESCRIPTION
**Pull request by devimc**
add steps to build kata kernel from source

Signed-off-by: Julio Montes <julio.montes@intel.com>

### CODE REVIEW

File: Developer-Guide.md [hunk 1]
```diff
@@ -227,18 +227,29 @@ $ (cd /usr/share/kata-containers && sudo ln -sf "$image" kata-containers-initrd.
...
+kernel_url="https://cdn.kernel.org/pub/linux/kernel/v$(echo $kernel_version | cut -f1 -d.).x/${kernel_tar_file}"
+curl -LOk ${kernel_url}
+tar -xf ${kernel_tar_file}
+mv .config "linux-${kernel_version}"
+pushd "linux-${kernel_version}"
+curl -L https://raw.githubusercontent.com/kata-containers/packaging/master/kernel/patches/0001-NO-UPSTREAM-9P-always-use-cached-inode-to-fill-in-v9.patch | patch -p1
+make ARCH=${kernel_arch} -j$(nproc)
+kata_kernel_dir="/usr/share/kata-containers"
+kata_vmlinuz="${kata_kernel_dir}/kata-vmlinuz-${kernel_version}.container"
+sudo cp -a "$(realpath arch/${kernel_arch}/boot/bzImage)" "${kata_vmlinuz}"
```
Comment (jodh-intel) [inline Developer-Guide.md]: These `cp` calls would probably be better as `install` calls with explicit user, group and perms options.

File: Developer-Guide.md [hunk 2]
```diff
@@ -227,18 +227,29 @@ $ (cd /usr/share/kata-containers && sudo ln -sf "$image" kata-containers-initrd.
...
-
-> **Note:**
->
-> - The files in the previous commands are from the Clear Containers
->   `linux-container` package. See [Assumptions](#assumptions).
-
-If you want to use an initrd image, install the [hyperstart](https://github.com/hyperhq/hyperstart/) kernel instead:
-```
-$ sudo wget -O /usr/share/kata-containers/vmlinuz.container https://github.com/hyperhq/hyperstart/raw/master/build/arch/x86_64/kernel
+kernel_arch="$(arch)"
```
Comment (jodh-intel) [inline Developer-Guide.md]: This new section means we can remove the details of the `linux-container` package in the "Assumptions" section I think?

**Comment by devimc** [comment]
@jodh-intel changes applied, thanks

File: Developer-Guide.md [hunk 3]
```diff
@@ -227,18 +226,29 @@ $ (cd /usr/share/kata-containers && sudo ln -sf "$image" kata-containers-initrd.
...
+tmpdir="$(mktemp -d)"
```
Comment (jodh-intel) [inline Developer-Guide.md]: I know it's a pita, but can you prefix these command lines with `$ ` please? :)

@klynnrif - ptal.

**Pull request merged**